whisper/whisperv5/doc.go

```diff
@@ -42,11 +40,11 @@ const (
 	ProtocolVersionStr = "5.0"
 	ProtocolName       = "shh"
 
-	statusCode           = 0
-	messagesCode         = 1
-	p2pCode              = 2
-	mailRequestCode      = 3
-	NumberOfMessageCodes = 32
+	statusCode           = 0 // used by whisper protocol
+	messagesCode         = 1 // normal whisper message
+	p2pCode              = 2 // peer-to-peer message (to be consumed by the peer, but not forwarded any futher)
+	p2pRequestCode       = 3 // peer-to-peer message, used by Dapp protocol
+	NumberOfMessageCodes = 64
```

Comment:
The reason for NumberOfMessageCodes being higher than the highest message code is forward compatibility concerns. I was planning to introduce new message types later (BloomFilterExchange, etc.), and AFAIU, if I change the NumberOfMessageCodes, the Whisper version must change as well, and it will be incompatible with the old version. It would be a big inconvenience to ask the entire network to upgrade just because of this minor change, which is otherwise backwards compatible. So, after discussion with Viktor we decided to reserve more message codes than currently necessary, and remove the unused ones. The current increase (64 vs. 32) was due to me feeling greedy -- I just wanted to grab additional 32 message codes for Whisper. The remaining 4 billion should be enough for eth, bzz, and any other possible protocol in the future.